test/functional/test_runner.py

```diff
@@ -401,16 +401,18 @@ def run_tests(*, test_list, src_dir, build_dir, tmpdir, jobs=1, enable_coverage=
     print_results(test_results, max_len_name, (int(time.time() - start_time)))
 
     if coverage:
-        coverage.report_rpc_coverage()
+        coverage_passed = coverage.report_rpc_coverage()
 
         logging.debug("Cleaning up coverage data")
         coverage.cleanup()
+    else:
+        coverage_passed = True
 
     # Clear up the temp directory if all subdirectories are gone
     if not os.listdir(tmpdir):
         os.rmdir(tmpdir)
 
-    all_passed = all(map(lambda test_result: test_result.was_successful, test_results))
+    all_passed = all(map(lambda test_result: test_result.was_successful, test_results)) and coverage_passed
```

Comment:
Mixed use of `coverage_passed`

might be more explicit to omit lines 408-409 and replace `coverage_passed` with `(coverage and coverage_passed)`

Reply:
I don't understand `(coverage and coverage_passed)` would always be false in a default run

Reply:
`(not coverage or coverage_passed)`